fix: Add parent constructor call in EnumReflectionProperty

jawira opened the pull request:
Fixes https://github.com/doctrine/persistence/issues/423

jawira committed:
fix: Add parent constructor call in EnumReflectionProperty

jawira commented:
Hi! is there anything I could do to help get this PR merged.  
All checks have passed.  
Thanks! :)

greg0ire commented:
> Once EnumReflectionProperty::__contruct calls its parent constructor (as in PR https://github.com/doctrine/persistence/pull/422) all methods like the following one can be safely removed (there's six of them):

Why don't you do it rightaway? Also, will there still be a point in keeping `private readonly ReflectionProperty $originalReflectionProperty` as a property if we do that?

I think this clashes a bit with the original design made in #248 by @IonBazan

jawira commented:
Ah you are right, I spoke too quickly. I cannot remove constructor since method's signature is not the same as parent class.  
However, I removed all "proxy methods" because they are not needed anymore :)

jawira commented:
Hello I have implemented all requested changes, could you accept this MR please ?

greg0ire commented:
It seems there are CI jobs failing. Please take a look at [this guide](https://www.doctrine-project.org/contribute/#dealing-with-checks-and-tools) for more on how to handle those.

greg0ire commented:
@IonBazan please review :pray:

IonBazan commented:
Thanks @jawira - LGTM. However, my changes were actually based on @beberlei https://github.com/doctrine/orm/pull/9304 so it's best to confirm once more with him.

Also good to add tests covering the changes you've made, reproducing the issue you reported.

jawira commented:
Hello @IonBazan, in **doctrine/orm**, `ReflectionEnumProperty` **DO** call the parent class.  
This change was made in https://github.com/doctrine/orm/pull/9381  
The code has evolved since, so I have updated the current PR to match the following:

https://github.com/doctrine/orm/blob/2c01dac17367c203de6fab0331c82eca0d639cdc/src/Mapping/ReflectionEnumProperty.php#L14-L27

I have also added tests to cover errors I had.

beberlei commented:
I am not sure anymore why I did it this way originally, but if this works here then great. However be mindful that for Doctrine ORM 3.4+ above we changed the abstraction completly and this code here becomes obsolete.

greg0ire commented:
Thanks @jawira !
